src/__test__/SearchPage.test.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import SearchPage from '../components/SearchPage';
+
+const mockStore = configureStore([]);
+
+describe('SearchPage component', () => {
+  test('SearchPage component renders correctly', () => {
+    const initialState = {
+      city: {
+        value: [
+          {
+            name: 'City Name',
+            lat: 11.1111,
+            lon: 22.2222,
+          },
+
+        ],
+      },
+    };
+
+    const store = mockStore(initialState);
+    const tree = renderer.create(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={['/search/City Name']}>
+          <SearchPage />
+        </MemoryRouter>
+      </Provider>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
- Kindly, some of your tests are failed please fix them 👍 
<img width="629" alt="2023-07-31_10-09" src="https://github.com/HtetWaiYan7191/weather-app/assets/52010789/16448f0f-bcb0-4033-a6ce-ac0a6d42bec1">

Reply:
Thank you very much for showing me the errors sir . I will update my tests and snapshots .